File: algorithms/D/DI String Match/DI String Match.js
var diStringMatch = function(s) {
    let i = 0, d = s.length, arr = [];
    
    for(let j = 0; j <= s.length; j += 1) {
        if(s[j] === 'I') arr.push(i++);
        else arr.push(d--);
    }
    
    return arr;
};

File: algorithms/D/DI String Match/DI String Match.cpp
class Solution {
public:
    vector<int> diStringMatch(string s) {
        vector<int> res;
    
        int I = 0;
        int D = s.size();
        
        for(int i=0;i<s.size()+1;i++)
        {
            if(s[i]=='I')
                res.push_back(I++);
            else
                res.push_back(D--);
        }
        return res;
    }
};
